orte/orted/orted_submit.c

```diff
@@ -3363,3 +3363,26 @@ void orte_profile_wakeup(int sd, short args, void *cbdata)
     /* abort the job */
     ORTE_ACTIVATE_JOB_STATE(NULL, ORTE_JOB_STATE_ALL_JOBS_COMPLETE);
 }
+
+int orte_submit_job_state_update(orte_job_t *jdata, orte_job_state_t state)
```

Comment:
nit: instead of passing in state, it could be checked at a higher level. It's fine as is though.

Reply:
Yeah I put in the state here because there is also a 'completing' callback that we could wire in. I didn't go that far in this PR since it wasn't needed for the bug fix and the side effects of that were unknown. I kept the function like this in case we need to wire it back in later - we can do so without adding a new function.